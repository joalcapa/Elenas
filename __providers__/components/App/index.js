import React from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {MockedProvider} from '@apollo/client/testing';

import config from '../../../src/kernel/v1/config';
import {responseGpql} from '../../../__factories__/gpql';

const App = (props) => {
    config.getInstance().setConfiguration({
        AsyncStorage,
        useUser: () => {
            return {
                user: null,
                userIsReady: true,
                updateUser: (newUser) => {
                    localStorage.setItem('USER', JSON.stringify(newUser));
                },
            }
        },
        env: {},
    });

    return (
        <MockedProvider mocks={responseGpql}>
            {props.children}
        </MockedProvider>
    );
}

export default App;