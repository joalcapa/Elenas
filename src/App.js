import './App.css';
import AsyncStorage from '@react-native-async-storage/async-storage';
import Kernel from './kernel';
import Navigation from './v1/navigation';

function App() {
  return (
    <Kernel storage={AsyncStorage}>
      <Navigation />
    </Kernel>
  );
}

export default App;
