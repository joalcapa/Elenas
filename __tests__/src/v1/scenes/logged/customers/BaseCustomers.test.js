import React from 'react';
import 'cross-fetch/polyfill';
import {create} from 'react-test-renderer';
import {View} from 'react-native';

import App from '../../../../../../__providers__/components/App';
import BaseCustomers from '../../../../../../src/v1/scenes/logged/customers/BaseCustomers';

const TextComponent = () => <View style={{width: 100, height: 100}} />;

describe('BaseCustomers component', () => {
    test('render TextComponent component without title and buttons', () => {
        const Component = BaseCustomers(TextComponent);
        const tree = create(<App><Component /></App>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render TextComponent component with title and without buttons', () => {
        const Component = BaseCustomers(TextComponent, 'Test scene');
        const tree = create(<App><Component /></App>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render TextComponent component with title and with create button', () => {
        const Component = BaseCustomers(TextComponent, 'Test scene', {isCreateButton: true});
        const tree = create(<App><Component /></App>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render TextComponent component with title and with list button', () => {
        const Component = BaseCustomers(TextComponent, 'Test scene', {isListButton: true});
        const tree = create(<App><Component /></App>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});