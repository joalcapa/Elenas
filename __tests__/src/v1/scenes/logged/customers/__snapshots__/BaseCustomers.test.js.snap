// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCustomers component render TextComponent component with title and with create button 1`] = `
<div
  className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
>
  <div
    className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
  >
    <div
      className="css-text-901oao r-fontWeight-vw2c0b"
      dir="auto"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Test scene
    </div>
    <div
      className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-17s6mgv r-paddingRight-1v0fv4f r-width-l0gqae"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-obd0qt r-display-6koalj r-paddingRight-1qfoi16 r-width-l0gwng"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-l7f4p8 r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
            dir="auto"
            onClick={[Function]}
          >
            CREAR CLIENTE
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-alignItems-obd0qt r-display-6koalj r-paddingRight-1qfoi16 r-width-l0gwng"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-l7f4p8 r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
            dir="auto"
            onClick={[Function]}
          >
            CERRAR SESION
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </div>
</div>
`;

exports[`BaseCustomers component render TextComponent component with title and with list button 1`] = `
<div
  className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
>
  <div
    className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
  >
    <div
      className="css-text-901oao r-fontWeight-vw2c0b"
      dir="auto"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Test scene
    </div>
    <div
      className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-17s6mgv r-paddingRight-1v0fv4f r-width-l0gqae"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-obd0qt r-display-6koalj r-paddingRight-1qfoi16 r-width-l0gwng"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-l7f4p8 r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
            dir="auto"
            onClick={[Function]}
          >
            LISTA DE CLIENTES
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-alignItems-obd0qt r-display-6koalj r-paddingRight-1qfoi16 r-width-l0gwng"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-l7f4p8 r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
            dir="auto"
            onClick={[Function]}
          >
            CERRAR SESION
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </div>
</div>
`;

exports[`BaseCustomers component render TextComponent component with title and without buttons 1`] = `
<div
  className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
>
  <div
    className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
  >
    <div
      className="css-text-901oao r-fontWeight-vw2c0b"
      dir="auto"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Test scene
    </div>
    <div
      className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-17s6mgv r-paddingRight-1v0fv4f r-width-l0gqae"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-obd0qt r-display-6koalj r-paddingRight-1qfoi16 r-width-l0gwng"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-l7f4p8 r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
            dir="auto"
            onClick={[Function]}
          >
            CERRAR SESION
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </div>
</div>
`;

exports[`BaseCustomers component render TextComponent component without title and buttons 1`] = `
<div
  className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
>
  <div
    className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
  >
    <div
      className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-17s6mgv r-paddingRight-1v0fv4f r-width-l0gqae"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-obd0qt r-display-6koalj r-paddingRight-1qfoi16 r-width-l0gwng"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-l7f4p8 r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
            dir="auto"
            onClick={[Function]}
          >
            CERRAR SESION
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-paddingLeft-b5h31w r-paddingTop-m611by"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </div>
</div>
`;
